### PR TITLE
Telink b91 I2C Driver Support

### PR DESCRIPTION
**Pull request by yurvyn**
I2C Driver implementation for Telink B91 SoC

### CODE REVIEW

**Comment by yurvyn** [comment]
Please approve the PR if you are OK with the changes.

**Comment by yurvyn** [comment]
Two approvals required.
Could anyone please approve the PR?

**Comment by yurvyn** [comment]
@kgugala, @pgielda
Since you are owners of most of the files in this PR, could you please approve it?
Thanks.

**Comment by yurvyn** [comment]
@galak, @kgugala, @nashif, @pgielda

There no any activity in this PR for 26 days already.
Could you please assign someone for this PR?

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
I noticed a few minor things, otherwise looks fine to me.

File: drivers/i2c/i2c_b91.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+#include <dt-bindings/pinctrl/b91-pinctrl.h>
+
+
+/* I2C configuration structure */
+struct i2c_b91_cfg {
+	uint32_t bitrate;
+	const uint32_t *pinctrl_list;
+	size_t pinctrl_list_size;
+};
+#define I2C_CFG(dev)                    ((struct i2c_b91_cfg *) ((dev)->config))
```
[the remark this answers is not in the document]
Comment (yurvyn) [reply drivers/i2c/i2c_b91.c]: Done.

File: drivers/i2c/i2c_b91.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+/* I2C data structure */
+struct i2c_b91_data {
+	struct k_sem mutex;
+};
+#define I2C_DATA(dev)                   ((struct i2c_b91_data *) ((dev)->data))
```
[the remark this answers is not in the document]
Comment (yurvyn) [reply drivers/i2c/i2c_b91.c]: Done.

File: drivers/i2c/i2c_b91.c [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(i2c_telink);
+
+#include <drivers/i2c.h>
+#include "i2c-priv.h"
+#include <drivers/pinmux.h>
+#include <dt-bindings/pinctrl/b91-pinctrl.h>
+
+
```
[the remark this answers is not in the document]
Comment (yurvyn) [reply drivers/i2c/i2c_b91.c]: Done.

File: drivers/i2c/i2c_b91.c [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+};
+#define I2C_CFG(dev)                    ((struct i2c_b91_cfg *) ((dev)->config))
+
+/* I2C data structure */
+struct i2c_b91_data {
+	struct k_sem mutex;
+};
+#define I2C_DATA(dev)                   ((struct i2c_b91_data *) ((dev)->data))
+
+
```
[the remark this answers is not in the document]
Comment (yurvyn) [reply drivers/i2c/i2c_b91.c:31]: Done.

File: drivers/i2c/i2c_b91.c [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+								  \
+	static struct i2c_b91_data i2c_b91_data_##inst;		  \
+								  \
+	static struct i2c_b91_cfg i2c_b91_cfg_##inst = {	  \
+		.bitrate = DT_INST_PROP(inst, clock_frequency),	  \
+		.pinctrl_list_size = ARRAY_SIZE(i2c_pins_##inst), \
+		.pinctrl_list = i2c_pins_##inst			  \
+	};							  \
+								  \
+	DEVICE_DT_INST_DEFINE(0, i2c_b91_init,			  \
```
[the remark this answers is not in the document]
Comment (yurvyn) [reply drivers/i2c/i2c_b91.c]: Done.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by yurvyn** [comment]
@carlescufi, @tbursztyka 

One more approval is left.
Could you please approver the PR?

**Comment by yurvyn** [comment]
Hello @cfriedt 

This PR is not assigned for more than 30 days, but already has 1 approval.
Could you please approve and merge it?

**Review by cfriedt — APPROVED** [review]

**Pull request merged**